docs/dedicated-data-endpoints/dedicated-data-endpoints.md

```diff
@@ -0,0 +1,132 @@
+<img src="./media/registry-private-link.png" width=500x>
+
+Unfortunately, VNet connectivity isn’t always an option.
+
+## Client Firewall Rules & Data Exfiltration Risks
+
+When connecting to a registry from on-prem hosts, IoT devices, custom build agents, or when Private Link may not be an option, client firewall rules may be applied, limiting access to specific resources.
+
+<img src="./media/registry-client-rules-all-storage.png" width=500x>
+
+As customers locked down their client firewall configurations, they realized they must create a rule with a wildcard for all storage accounts, raising concerns for data-exfiltration. A bad actor could deploy code that would be capable of writing to another "Evil-Empire" storage account.
+
+To mitigate data-exfiltration concerns, Azure Container Registry is making dedicated data-endpoints available.
+
+<img src="./media/registry-data-exfiltration.png" width=500x>
+
+## Dedicated Data-endpoints
+
+When dedicated data-endpoints are enabled, layers are retrieved from the Azure Container Registry service, with fully qualified domain names representing the registry domain. As any registry may become geo-replicated, a regional pattern is used: 
+
+**[registry]**`.`**[region]**`.data.azurecr.io`. 
+
+For the Contoso example, multiple regional data-endpoints are added supporting the local region with a nearby replica.
+
+With dedicated data-endpoints, the bad actor is blocked from writing to other storage accounts.
+
+<img src="./media/registry-dedicated-data-endpoint.png" width=500x>
+
+## Enabling Dedicated Data-endpoints
+
+**Note:** Switching to dedicated data-endpoints will impact clients that have configured firewall access to the existing `*.blob.core.windows.net` endpoints, causing pull failures. To assure clients have consistent access, add the new data-endpoints to the client firewall rules. Once completed, existing registries can enable dedicated data-endpoints through the `az cli`, or the Azure portal.
+
+### Private Preview Configuration
+
+Until the Portal and az cli are enabled, customers can use the `az rest` api to enable dedicated data-endpoints.
+
+- Set the registry default
+
+  ```sh
+  az configure --defaults acr=demo42
+  ```
+
+- Export the resource id of the registry
+
+  ```sh
+  export RESOURCE_ID=$(az acr show --query id -o tsv)
+  ```
+
+- Execute the REST api with the `az rest` command
+
+  ```sh
+  az rest --method patch --uri "$RESOURCE_ID?api-version=2019-12-01-preview" --body "{ \"properties\":{\"dataEndpointEnabled\":true}}" -o json
+  ```
+
+  Look for:
+
+  ```json
+  "dataEndpointEnabled": true,
+      "dataEndpointHostNames": [
+        "demo42.eastus.data.azurecr.io",
+        "demo42.westus.data.azurecr.io"
+  ```
+
+### az CLI
+
+Using [az cli](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest) version 2.3.1 or greater, run the [az acr update](https://docs.microsoft.com/en-us/cli/azure/acr?view=azure-cli-latest#az-acr-update) command:
+
```

Comment:
CLI version OK? I didn't find the `--data-endpoint-enabled` in my Mac 2.3.1 installation.